### PR TITLE
Hybrid HTTP connection handles URL requests with spaces. 

### LINKED ISSUE #82: Request with encoded spaces are not handled correctly
The listener fails to handle URLs with spaces. The client sends the request properly encoded but the hybrid HTTP connection fails to parse the URL and throws an exception.  

```java

java.util.concurrent.CompletionException: java.lang.RuntimeException: java.net.URISyntaxException: Illegal character in path at index 68: sb://test.servicebus.windows.net/echo/api/contents/Untitled Folder?type=directory&_=1644956591273
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1739)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1728)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.RuntimeException: java.net.URISyntaxException: Illegal character in path at index 68: sb://test.servicebus.windows.net/echo/api/contents/Untitled Folder?type=directory&_=1644956591273
	at com.microsoft.azure.relay.HybridHttpConnection.invokeRequestHandler(HybridHttpConnection.java:173)
	at com.microsoft.azure.relay.HybridHttpConnection.lambda$processFirstRequestAsync$2(HybridHttpConnection.java:115)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
	... 6 common frames omitted
Caused by: java.net.URISyntaxException: Illegal character in path at index 68: sb://test.servicebus.windows.net/echo/api/contents/Untitled Folder?type=directory&_=1644956591273
	at java.base/java.net.URI$Parser.fail(URI.java:2913)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3084)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3166)
	at java.base/java.net.URI$Parser.parse(URI.java:3114)
	at java.base/java.net.URI.<init>(URI.java:600)
	at com.microsoft.azure.relay.HybridHttpConnection.invokeRequestHandler(HybridHttpConnection.java:170)
	... 8 common frames omitted
```

### PR DESCRIPTION
**Pull request by giventocode**
The path part of the request URI is encoded using jetty ```URIUtil.encodePath``` - which is already a dependency in the SDK. 

Closes #82

### CODE REVIEW

File: src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java [hunk 1]
```diff
@@ -208,6 +205,23 @@ void invokeRequestHandler(RequestCommandAndStream requestAndStream) {
 		}
 	}
 
+	private URI parseRequestUri(URI listenerAddress, String requestTarget) throws URISyntaxException {
```
Comment (bainian12345) [inline src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java:208]: nit: maybe rename to `parseAndEncodeRequestUri`?

Comment (bainian12345) [inline src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java:208]: could be static

File: src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java [hunk 2]
```diff
@@ -315,7 +316,57 @@ public void parseRequestPathAndQueryTest() throws IOException, URISyntaxExceptio
...
-	
+
+	@Test
+	public void parseRequestPathWithSpacesAndQueryTest() throws IOException {
+		CompletableFuture<Void> requestReceived = new CompletableFuture<Void>();
+		String extraPath = "/extra Path/extra  Path2";
+		String extraQuery = "queryKey=queryValue&queryKey2=queryValue;"; // mixing in a special char just for fun
+
+		listener.setRequestHandler((context) -> {
+			try {
```
Comment (bainian12345) [inline src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java:327]: Just for completeness, would you mind setting and returning a response inside the handler to make sure that works as well?

File: src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java [hunk 3]
```diff
@@ -315,7 +316,57 @@ public void parseRequestPathAndQueryTest() throws IOException, URISyntaxExceptio
...
+		urlBuilder.append("?").append(extraQuery);
+
+		URL url = new URL(urlBuilder.toString());
+		String tokenString = tokenProvider.getTokenAsync(url.toString(), Duration.ofHours(1)).join().getToken();
+
+		HttpURLConnection conn = (HttpURLConnection)url.openConnection();
+		conn.setRequestMethod("GET");
+		conn.setRequestProperty("ServiceBusAuthorization", tokenString);
+
+		conn.getResponseCode();
```
Comment (bainian12345) [inline src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java:361]: Would be great if you can add a check to verify the response code and message as well. The Relay service may still return internal server error or something else.

**Review by bainian12345 — CHANGES_REQUESTED** [review]
:clock1:

**Review by bainian12345 — APPROVED** [review]
:shipit:

**Comment by Qi77Qi** [comment]
hi @bainian12345 will this fix be released soon?

**Pull request merged**